zipkin-storage/zipkin2_cassandra/src/main/java/zipkin2/storage/cassandra/DefaultSessionFactory.java

```diff
@@ -55,12 +58,15 @@
       Cluster cluster = closer.register(buildCluster(cassandra));
       cluster.register(new QueryLogger.Builder().build());
       Session session;
+      String keyspace = cassandra.keyspace();
       if (cassandra.ensureSchema()) {
         session = closer.register(cluster.connect());
-        Schema.ensureExists(cassandra.keyspace(), cassandra.searchEnabled(), session);
-        session.execute("USE " + cassandra.keyspace());
+        Schema.ensureExists(keyspace, cassandra.searchEnabled(), session);
+        Schema.ensureExists(DEFAULT_KEYSPACE + "_udts", false, session);
```

Comment:
Just out of curiosity, when ensureSchema is enabled, aren't you creating 2 keyspaces here?